src/Mobs/Path.cpp

```diff
@@ -51,6 +51,14 @@ cPath::cPath(
 	m_Source.x = FloorC(a_StartingPoint.x - HalfWidthInt);
 	m_Source.y = FloorC(a_StartingPoint.y);
 	m_Source.z = FloorC(a_StartingPoint.z - HalfWidthInt);
+	
+	if ((m_Source.y <= 0) || (m_Destination.y <= 0)
+		|| (m_Source.y > 256) || (m_Destination.y > 256))
```

Comment:
I don't think that would be proper, that's making the function do something that it isn't supposed to do.

Instead, `IsSolid` should assert that the coords given to it are valid, and anything that calls `IsSolid` should make sure the coords are valid. Ideally, paths for invalid coords shouldn't be created at all.